app/script/conversation/ConversationCellState.js

```diff
@@ -23,279 +23,282 @@ window.z = window.z || {};
-
-        switch (activity) {
-          case 'message':
-            if (count === 1) {
-              activity_strings.push(z.l10n.text(z.string.conversationsSecondaryLineNewMessage, count));
-            } else if (count > 1) {
-              activity_strings.push(z.l10n.text(z.string.conversationsSecondaryLineNewMessages, count));
-            }
-            break;
-          case 'ping':
-            if (count === 1) {
-              activity_strings.push(z.l10n.text(z.string.conversationsSecondaryLinePing, count));
-            } else if (count > 1) {
-              activity_strings.push(z.l10n.text(z.string.conversationsSecondaryLinePings, count));
-            }
-            break;
-          case 'call':
-            if (count === 1) {
-              activity_strings.push(z.l10n.text(z.string.conversationsSecondaryLineMissedCall, count));
-            } else if (count > 1) {
-              activity_strings.push(z.l10n.text(z.string.conversationsSecondaryLineMissedCalls, count));
-            }
-            break;
-          default:
-        }
-      }
-    }
-
-    return activity_strings.join(', ');
-  }
+  const ACTIVITY_TYPE = {
+    CALL: 'ConversationCellState.ACTIVITY_TYPE.CALL',
+    MESSAGE: 'ConversationCellState.ACTIVITY_TYPE.MESSAGE',
+    PING: 'ConversationCellState.ACTIVITY_TYPE.PING',
+  };
 
-  function accumulate_activity(conversation_et) {
-    const unread_events = conversation_et.unread_events();
+  const _accumulateActivity = conversationEntity => {
+    const unreadEvents = conversationEntity.unread_events();
     const activities = {
-      call: 0,
-      message: 0,
-      ping: 0,
+      [ACTIVITY_TYPE.CALL]: 0,
+      [ACTIVITY_TYPE.MESSAGE]: 0,
+      [ACTIVITY_TYPE.PING]: 0,
     };
 
-    for (const message_et of unread_events) {
-      if (message_et.is_call() && message_et.was_missed()) {
-        activities.call = activities.call + 1;
-      } else if (message_et.is_ping()) {
-        activities.ping = activities.ping + 1;
-      } else if (message_et.is_content()) {
-        activities.message = activities.message + 1;
+    for (const messageEntity of unreadEvents) {
+      if (messageEntity.is_call() && messageEntity.was_missed()) {
+        activities[ACTIVITY_TYPE.CALL] += 1;
+      } else if (messageEntity.is_ping()) {
+        activities[ACTIVITY_TYPE.PING] += 1;
+      } else if (messageEntity.is_content()) {
+        activities[ACTIVITY_TYPE.MESSAGE] = +1;
       }
     }
 
-    return generate_activity_string(activities);
-  }
+    return _generateActivityString(activities);
+  };
 
-  const default_state = {
-    description() {
-      return '';
-    },
-    icon() {
-      return z.conversation.ConversationStatusIcon.NONE;
-    },
+  const _generateActivityString = activities => {
+    return Object.keys(activities)
+      .map(activity => {
+        const activityCount = activities[activity];
+        const hasActivity = activityCount >= 1;
+
+        if (hasActivity) {
```

Comment:
The following is still very clear.. since the `activityCount` is a number
```js
if (activityCount) {
```

Reply:
Very true, Ende like this after multiple refactorings as we did the check for each case in the switch individually.